CSharp/ArticleProjects/DotNetTutorial/Program.cs

```diff
@@ -286,17 +286,33 @@ private static string GetContainerSasUrl(CloudBlobClient blobClient, string cont
         /// <param name="resourceFiles">A collection of <see cref="ResourceFile"/> objects representing blobs within
         /// a Storage account container. The StartTask will download these files from Storage prior to execution.</param>
         /// <returns>A <see cref="System.Threading.Tasks.Task"/> object that represents the asynchronous operation.</returns>
-        private static async Task CreatePoolAsync(BatchClient batchClient, string poolId, IList<ResourceFile> resourceFiles)
+        private static async Task CreatePoolIfNotExistAsync(BatchClient batchClient, string poolId, IList<ResourceFile> resourceFiles)
         {
-            Console.WriteLine("Creating pool [{0}]...", poolId);
-
-            // Create the unbound pool. Until we call CloudPool.Commit() or CommitAsync(), no pool is actually created in the
-            // Batch service. This CloudPool instance is therefore considered "unbound," and we can modify its properties.
-            CloudPool pool = batchClient.PoolOperations.CreatePool(
-                poolId: poolId,
-                targetDedicated: 3,                                                         // 3 compute nodes
-                virtualMachineSize: "small",                                                // single-core, 1.75 GB memory, 225 GB disk
-                cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));   // Windows Server 2012 R2
+            CloudPool pool = null;
+            try
+            {
+                Console.WriteLine("Creating pool [{0}]...", poolId);
+
+                // Create the unbound pool. Until we call CloudPool.Commit() or CommitAsync(), no pool is actually created in the
+                // Batch service. This CloudPool instance is therefore considered "unbound," and we can modify its properties.
+                pool = batchClient.PoolOperations.CreatePool(
+                    poolId: poolId,
+                    targetDedicated: 3,                                                         // 3 compute nodes
+                    virtualMachineSize: "small",                                                // single-core, 1.75 GB memory, 225 GB disk
+                    cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));   // Windows Server 2012 R2

```

Comment:
Basically move this code up into the try as well:
```
            // Create and assign the StartTask that will be executed when compute nodes join the pool.
            // In this case, we copy the StartTask's resource files (that will be automatically downloaded
            // to the node by the StartTask) into the shared directory that all tasks will have access to.
            pool.StartTask = new StartTask
            {
                // Specify a command line for the StartTask that copies the task application files to the
                // node's shared directory. Every compute node in a Batch pool is configured with a number
                // of pre-defined environment variables that can be referenced by commands or applications
                // run by tasks.

                // Since a successful execution of robocopy can return a non-zero exit code (e.g. 1 when one or
                // more files were successfully copied) we need to manually exit with a 0 for Batch to recognize
                // StartTask execution success.
                CommandLine = "cmd /c (robocopy %AZ_BATCH_TASK_WORKING_DIR% %AZ_BATCH_NODE_SHARED_DIR%) ^& IF %ERRORLEVEL% LEQ 1 exit 0",
                ResourceFiles = resourceFiles,
                WaitForSuccess = true
            };

await pool.CommitAsync();
```